apigentools/commands/push.py

```diff
@@ -3,72 +3,46 @@
 # This product includes software developed at Datadog (https://www.datadoghq.com/).
 # Copyright 2019 Datadog, Inc.
 import logging
-import os
 import tempfile
 import time
+import subprocess
 
 from apigentools.commands.command import Command
 from apigentools.utils import change_cwd, run_command
 
 log = logging.getLogger(__name__)
 
-REPO_SSH_URL = 'git@github.com:{}/{}.git'
-REPO_HTTPS_URL = 'https://github.com/{}/{}.git'
 
 class PushCommand(Command):
 
     def run(self):
-        log.error("The push command is a work in progress and isn't currently available")
-        return 1
-
         created_branches = {}
         cmd_result = 0
 
-        versions = self.args.api_versions or self.config.spec_versions
         languages = self.args.languages or self.config.languages
 
         for lang_name, lang_config in self.config.language_configs.items():
             # Skip any languages not specified by the user
             if lang_name not in languages:
                 continue
-            # Note: We assume that its one repo for all versions.
-            # Git shallow clone the repository into a temp dir
-            with tempfile.TemporaryDirectory() as temp_repo_dir:
-                # Choose to use HTTPS or SSH method of cloning the repo
-                # Defaults to SSH
-                if self.args.use_https:
-                    repo = REPO_HTTPS_URL.format(lang_config.github_org, lang_config.github_repo)
-                else:
-                    repo = REPO_SSH_URL.format(lang_config.github_org, lang_config.github_repo)
 
+            gen_dir = self.get_generated_lang_dir(lang_name)
+
+            # Assumes all generated changes are in the gen_dir directory
+            # This is done by default in the `generate` command.
+            with change_cwd(gen_dir):
+                repo = "{}/{}".format(lang_config.github_org, lang_config.github_repo)
+                branch_name = '{}/{}'.format(lang_name, time.time())
                 try:
-                    run_command(['git', 'clone', '--depth=2', repo, temp_repo_dir])
+                    run_command(['git', 'checkout', '-b', branch_name])
+                    run_command(['git', 'add', '-A'])
+                    run_command(['git', 'commit', '-a', '-m', "'Pushing latest generated client'"])
```

Comment:
* Let's make the comment modifiable by an argument.
* Let's make the default comment be something like "Regenerate from commit abcd123 of spec repo"

Reply:
Hmm, I see that you added `--push-commit-mgs` argument, but you forgot to utilize it here.

Reply:
🤦‍♂